tests/assemble.py

```diff
@@ -53,6 +53,30 @@ def test_parse_line():
     assert a.parse_line(next(lines)) == (None, '.data', ())  # test left-aligned directive is not treated as label
 
 
+def test_parse_labels_correctly():
+    """
+    description of what defines a label
+    https://sourceware.org/binutils/docs/as/Statements.html
+    https://sourceware.org/binutils/docs/as/Labels.html
+    """
+    a = Assembler()
+    assert a.parse_line('label: .set const, 42') == ('label', '.set', ('const', '42',))
+    assert a.parse_line('label:.set const, 42') == ('label', '.set', ('const', '42',))
+    assert a.parse_line('label:') == ('label', None, ())
+    assert a.parse_line('    label:') == ('label', None, ())
+    assert a.parse_line('    label:  ') == ('label', None, ())
```

Comment:
are blanks left of the label allowed by the assembler syntax?

Reply:
Yes. That is allowed, according to https://sourceware.org/binutils/docs/as/Statements.html:
> Whitespace before a label or after a colon is permitted, but you may not have whitespace between a label’s symbol and its colon